cmd/dagger/module.go

```diff
@@ -732,6 +732,9 @@ func getModuleConfigurationForSourceRef(
 				return nil, fmt.Errorf("failed to unmarshal %s: %s", configPath, err)
 			}
 			if namedDep, ok := modCfg.DependencyByName(srcRefStr); ok {
+				if strings.HasPrefix(namedDep.Source, "github.com/") {
```

Comment:
Bummer to have to special-case this, what broke with the recursive approach?

Reply:
It's possible that it was fine and I hit a flaky test, but my fear was something with special cases with nested modules. Based on the timing of the test execution I was worried that it hit an infinite loop

Reply:
https://github.com/dagger/dagger/actions/runs/7893082590/job/21540933271

Reply:
We could potentially check
```go
conf.Source = dag.ModuleSource(namedDep.Source)
conf.SourceKind, err = conf.Source.Kind(ctx)
if conf.SourceKind == dagger.GitSource
 // recurse
```
instead of specifically looking for `github.com/`

Reply:
> We could potentially check instead of specifically looking for `github.com/`

That sounds good to me :tada: